### PR TITLE
update the dependencies to react 18

### PR DESCRIPTION
**Pull request by ududsha**
When we use this library within another project with the React 18 env we are getting below peer dependency error.

So i have upgrade the dependencies here to React 18.

Test using the commands provided and all worked fine. Mostly React 18 upgrade affect the way we use react-dom. Read more [here](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html).

### CODE REVIEW

**Comment by ududsha** [comment]
Hi @KingSora Please help check this PR. This would solve some dependency issues which caused due to React 18+ upgrades when OverlayScrollbars is used in that project.

File: packages/overlayscrollbars-react/package.json [hunk 1]
```diff
@@ -39,8 +39,8 @@
         "typescript": "^3.8.3"
     },
     "peerDependencies": {
-        "react": "^16.4.0 || ^17.0.0",
-        "overlayscrollbars": "^1.10.0"
+        "overlayscrollbars": "^1.10.0",
+        "react": "^18.0.0"
```
Comment (KingSora) [inline packages/overlayscrollbars-react/package.json:43]:
```suggestion
        "react": "^16.4.0 || ^17.0.0 || ^18.0.0"
```

To keep the compatibility to older react versions

**Comment by KingSora** [comment]
@ududsha Can you explain why so many files are changed? To my understanding only the `package.json` (or `package-lock.json`) should be changed here.

**Comment by ududsha** [comment]
@KingSora I think my formatter ran though some files.

**Pull request closed without merging**